pages/[id]/shop/toy.js

```diff
@@ -0,0 +1,87 @@
+import StyledLeftButton from "@/components/StyledComponents/StyledLeftButton";
+import { useRouter } from "next/router";
+import { useState } from "react";
+import ShopTable from "@/components/Shop/ShopTable";
+import { toys } from "@/lib/shop.js";
+import MoneyCounter from "@/components/util/MoneyCounter";
+import StyledDefaultHeader from "@/components/StyledComponents/StyledDefaultHeader";
+import ConfirmationPopup from "@/components/util/ConfirmPopUp";
+
+export default function ToyShop({
+  userStats,
+  onUpdateInventoryToy,
+  onSubtractMoney,
+}) {
+  const [selectedToyId, setSelectedToyId] = useState(null);
+  const [itemCost, setItemCost] = useState(0);
+  const router = useRouter();
+  const { id } = router.query;
+
+  function isToyPurchased(id) {
+    for (const toy of userStats.inventory.toy) {
+      if (toy.id === id) {
+        return toy.purchased;
+      }
+    }
+    return false;
+  }
```

Comment:
I'm not 100% sure (I haven't tested it in code), but I think the following approach with the Array Method [`some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) leads to the same result and is a little easier to understand. 
```suggestion
 function isToyPurchased(id) {
    return userStats.inventory.toy.some(toy => toy.id === id && toy.purchased);
  }

```